import-export-cli/integration/testutils/policy_testUtils.go

```diff
@@ -0,0 +1,329 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package testutils
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/wso2/product-apim-tooling/import-export-cli/integration/apim"
+	"github.com/wso2/product-apim-tooling/import-export-cli/integration/base"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	policyIDKey   = "policyId"
+	policyNameKey = "policyName"
+	policyIDReset = "-1"
+
+	applicationPolicySubType  = "application policy"
+	advancedPolicySubType     = "advanced policy"
+	customPolicySubType       = "custom rule"
+	subscriptionPolicySubType = "subscription policy"
+)
+
+// ValidateThrottlePolicyExportImport : Validates Exporting Throttling Policy from source env and Importing to destination env
+func ValidateThrottlePolicyExportImport(t *testing.T, args *PolicyImportExportTestArgs, adminUsername, adminPassword, policyType string) {
+	t.Helper()
+
+	// Setup apictl envs
+	base.SetupEnv(t, args.SrcAPIM.GetEnvName(), args.SrcAPIM.GetApimURL(), args.SrcAPIM.GetTokenURL())
+	base.SetupEnv(t, args.DestAPIM.GetEnvName(), args.DestAPIM.GetApimURL(), args.DestAPIM.GetTokenURL())
+
+	// Export policy from env 1
+	base.Login(t, args.SrcAPIM.GetEnvName(), args.CtlUser.Username, args.CtlUser.Password)
+	policyName := fmt.Sprintf("%v", args.Policy[policyNameKey])
+
+	exportedOutput, err := exportThrottlePolicy(t, policyName, args.SrcAPIM.GetEnvName(), args.Type)
+	assert.Nil(t, err, "Error while exporting the Throttling Policy")
+
+	args.ImportFilePath = base.GetExportedPathFromOutput(exportedOutput)
+	assert.True(t, base.IsFileAvailable(t, args.ImportFilePath))
+	args.SrcAPIM.DeleteThrottlePolicy(fmt.Sprintf("%v", args.Policy[policyIDKey]), policyType)
+
+	// Import Throttling Policy to env 2
+	base.Login(t, args.DestAPIM.GetEnvName(), args.CtlUser.Username, args.CtlUser.Password)
+	importedOutput, err := importThrottlePolicy(t, args)
+	assert.Nil(t, err, "Error while importing the Throttling Policy")
+	assert.Contains(t, importedOutput, "Successfully imported")
+	// Give time for newly imported Throttling Policy to get indexed
+	base.WaitForIndexing()
+
+	// Get Throttle Policy from env 2
+	importedPolicy, _ := getThrottlingPolicyByName(t, args, adminUsername, adminPassword, policyName, policyType)
+	// Validate env 1 and env 2 policy is equal
+	validatePoliciesEqual(t, args, importedPolicy)
+	removeExportedThrottlingPolicyFile(t, args.ImportFilePath)
+}
+
+// ValidateThrottlePolicyImportUpdate : Validates importing existing Throttling Policy
+func ValidateThrottlePolicyImportUpdate(t *testing.T, args *PolicyImportExportTestArgs, adminUsername, adminPassword, policyType string) {
```

Comment:
The behaviour of this function is not exactly correct. Here, we export an API and import it with the update flag. But the real use case would be, export the API, import it, and reimport it with the update flag.

Otherwise, according to the behaviour of this function, only a first time import will happen always even though we specify the update flag. We need to change the behaviour.

Reply:
Refer [1].

[1] https://github.com/wso2/product-apim-tooling/blob/5fce77894c0117a8e45239e3f6abaa36034d8248/import-export-cli/integration/testutils/apiProduct_testUtils.go#L338

Reply:
fixed with 19f5574e5c4f3995d95904eaccfa0c98038dd749